src/client/socksclient.ts

```diff
@@ -311,6 +311,9 @@ class SocksClient extends EventEmitter implements SocksClient {
         this._options.proxy.port,
         this._options.proxy.ipaddress
       );
+      if (this._options.set_tcp_nodelay !== undefined && this._options.set_tcp_nodelay !== null) {
```

Comment:
We can probably just have this check and I think everything else looks good.
```suggestion
      if (this._options.set_tcp_nodelay) {
```

Reply:
I thought about that as well but then if it's explicitly set to `false` and the default case of `false` changes then it would break.

Reply:
But it could be simplified to `... != null` which is equal to `... !== undefined && ... !== null` but I don't mind the longer form either as it's more explicit in what it does.

Reply:
That was my thinking behind it, either `null` or `undefined` is default, other types (especially boolean) override the default value. Can we keep it this way or shall I change it as @talmobi suggests - use `... != null` ?

Reply:
Let's just leave it as-is.